src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/ServiceToCcdServiceProcessor.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.locationrefdata.camel.processor;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.Exchange;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.data.ingestion.camel.exception.RouteFailedException;
+import uk.gov.hmcts.reform.data.ingestion.camel.processor.JsrValidationBaseProcessor;
+import uk.gov.hmcts.reform.data.ingestion.camel.validator.JsrValidatorInitializer;
+import uk.gov.hmcts.reform.locationrefdata.camel.binder.ServiceToCcdCaseType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static java.util.stream.Stream.of;
+
+@Slf4j
+@Component
+public class ServiceToCcdServiceProcessor extends JsrValidationBaseProcessor<ServiceToCcdCaseType> {
+
+    @Autowired
+    JsrValidatorInitializer<ServiceToCcdCaseType> serviceToCcdServiceJsrValidatorInitializer;
+
+    @Value("${logging-component-name}")
+    private String logComponentName;
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void process(Exchange exchange) throws Exception {
+
+        List<ServiceToCcdCaseType> serviceToCcdCaseTypes;
+
+        serviceToCcdCaseTypes = exchange.getIn().getBody() instanceof List
+            ? (List<ServiceToCcdCaseType>) exchange.getIn().getBody()
+            : singletonList((ServiceToCcdCaseType) exchange.getIn().getBody());
+
+        log.info(" {} ServiceToCCDService Records count before Validation & before merging service names {}::",
+                 logComponentName, serviceToCcdCaseTypes.size()
+        );
+
+        List<ServiceToCcdCaseType> refinedServiceToCcdCaseTypes = populateService(serviceToCcdCaseTypes);
+        log.info(" {} ServiceToCCDService Records count before Validation & after merging service names {}::",
+                 logComponentName, refinedServiceToCcdCaseTypes.size()
+        );
+
+        List<ServiceToCcdCaseType> filteredServiceToCcdCaseTypes = validate(
+            serviceToCcdServiceJsrValidatorInitializer,
+            refinedServiceToCcdCaseTypes
+        );
+        log.info(" {} ServiceToCCDService Records count after Validation & after merging service names {}::",
+                 logComponentName, filteredServiceToCcdCaseTypes.size()
+        );
+
+        audit(serviceToCcdServiceJsrValidatorInitializer, exchange);
+
+        if (filteredServiceToCcdCaseTypes.size() == 0) {
+            log.error(" {} ServiceToCcdService failed as no valid records present::", logComponentName);
+
+            throw new RouteFailedException("ServiceToCcdService failed as no valid records present");
+        }
+        exchange.getMessage().setBody(filteredServiceToCcdCaseTypes);
```

Comment:
looks like we are not handling scenarios where ccd case type can be null in file
There are three possibilities and all should be handled -

1. All columns have value in csv
2. Some of the rows dont have ccd case type. In this case ccd case type should be blank in table
3. Some rows dont have ccd service name and case type. In this case it should be ignored and no record should be inserted. If there is already a record for service code then it should be deleted from the table

Reply:
done